### PR TITLE
publishing Shiny R lesson

### PR DESCRIPTION
**Pull request by hawc2**
@svmelton @yann-ryan do you two mind reviewing and approving this pull request to publish Yann's lesson on Shiny? Anisa is out on leave but I think I did everything necessary for the JISC banner and all the markdown should render right, but please do a thorough look through to verify the lesson looks good to go. Thanks!

Relevant lesson issue [416](https://github.com/programminghistorian/ph-submissions/issues/416) plus the [translation issue ticket](https://github.com/programminghistorian/ph-submissions/issues/486)

### Checklist 

- [x] Assign yourself in the "Assignees" menu
- [x] Assign at least one individual or team to "Reviewers"
  - [x] if the text needs to be translated, assign the relevant language team(s) as "Reviewers" and tag both the team as well as the managing edtor in your PR. Please follow the [translation request guidelines](https://github.com/programminghistorian/jekyll/wiki/Requesting-Translation-Guidelines) when writing your PR description
- [x] Add the appropriate "Label"
- [x] [Ensure the status checks pass](https://github.com/programminghistorian/jekyll/wiki/Making-Technical-Contributions#4-check-your-pr-status)
- [x] [Check the live preview of your PR on Netlify](https://github.com/programminghistorian/jekyll/wiki/Making-Technical-Contributions#5-preview-how-your-pr-looks-when-built-into-html)
- [x] If this PR closes an open issue, add the phrase `Closes #ISSUENUMBER` to the description above

*If you are having difficulty fixing build errors, first consult <https://github.com/programminghistorian/jekyll/wiki/Making-Technical-Contributions> carefully, especially ["Common Build Errors"](https://github.com/programminghistorian/jekyll/wiki/Making-Technical-Contributions#common-build-errors). Then contact the technical team if you need further help.*

### CODE REVIEW

**Comment by hawc2** [comment]
@jenniferisasi also adding you here for review since you're working on a translation of this lesson

File: en/lessons/shiny-leaflet-newspaper-map-tutorial.md [hunk 1]
```diff
@@ -0,0 +1,505 @@
...
+## Lesson Aims
+
+In this lesson, you will learn:
+
+-   How to create a basic interactive Shiny application.
+-   The key layouts and design principles of the Shiny UI.
+-   The concept and practice of 'reactive programming', as implemented by Shiny applications. Specifically, you'll learn how you can use Shiny to 'listen' for certain inputs, and how they are connected to outputs displayed in your app.
+
+<div class="alert alert-info">
+Note that this lesson doesn't teach any coding in R, other than what's necessary to create the web application, nor does it cover publishing the finished application to the web. A basic knowledge of R, particularly using the [tidyverse](/en/lessons/data_wrangling_and_management_in_R), would be very useful.
```
Comment (svmelton) [inline en/lessons/shiny-leaflet-newspaper-map-tutorial.md]: The markdown in this banner isn't rendering 
Screen Shot 2022-10-18 at 10 02 59 PM

File: en/lessons/shiny-leaflet-newspaper-map-tutorial.md [hunk 2]
```diff
@@ -0,0 +1,505 @@
...
+
+### Create an Empty Shiny Application  
+
+A Shiny application consists of a script file with a special, reserved filename, `app.R`, which tells R Studio to treat that script as an application and open it in a web browser when it is run. In this first section, you will create an application which will load the relevant libraries and datasets, and display a test 'Hello World' message. To do this, carry out the following steps:  
+
+1. Set up an Application Folder
+
+It's good practice to put all the necessary files for the application in their own folder, within the RStudio project. Do this by creating a new folder called 'newspaper-app' within the folder of the RStudio project you just created. Place the files you downloaded above (`BritishAndIrishNewspapersTitleList_20191118.csv` and `newspaper_coordinates.csv`) into this new folder.
+
+2. Make the app.R file.
```
Comment (svmelton) [inline en/lessons/shiny-leaflet-newspaper-map-tutorial.md]: Small issue, but all the numbers are rendering as "1"
Screen Shot 2022-10-18 at 10 18 54 PM

File: en/lessons/shiny-leaflet-newspaper-map-tutorial.md [hunk 3]
```diff
@@ -0,0 +1,505 @@
...
+```
+ui = fluidPage(
+
+"Hello World"
+
+    )
+```    
+
+
+2. Create the Server Element
```
Comment (svmelton) [inline en/lessons/shiny-leaflet-newspaper-map-tutorial.md]: Same issue as above re: numbering

File: en/lessons/shiny-leaflet-newspaper-map-tutorial.md [hunk 4]
```diff
@@ -0,0 +1,505 @@
...
+
+**Outputs** are the R expressions which tell Shiny what to display in the UI and are created, in the server, with the variable name `output$*`. Outputs need to be matched to a UI `*Output` element. In the UI, you created a `leaflet` output with the label `map` using the code `leafletOutput('map')`. This should be matched to an output in the server named `output$map`.
+
+In turn, this `output$map` variable should be set to a Shiny `render*` function, which tells Shiny what kind of object is to be rendered in the UI. The one we need is called `renderLeaflet`, which tells the UI to output a map created by the `leaflet` library. The `renderLeaflet` object has both parentheses and curly braces, just like the reactive object we created above.
+
+The Leaflet map itself will be created within this. First, add the function `leaflet()`. Next, add the default tiles (the zoomable map images) using `addTiles()`. Next, set the default map position and zoom to Britain and Ireland using the command `setView(lng = -5, lat = 54, zoom = 5)`.
+
+###  Draw Points using the Reactive Dataframe.
```
Comment (svmelton) [inline en/lessons/shiny-leaflet-newspaper-map-tutorial.md]: Delete the period

**Review by svmelton — CHANGES_REQUESTED** [review]
Most of the comments here are minor (one typo, a couple of issues with numbering) and one of the banners isn't displaying correctly—the markdown isn't rendering.

**Comment by hawc2** [comment]
Thanks for catching all these details @svmelton! @yann-ryan will next do a final look through and adjust these markdown issues

**Review by yann-ryan — APPROVED** [review]

**Comment by hawc2** [comment]
@svmelton this pull request should have the changes you asked for, can you take a look now and approve or let us know about further edits?

@jenniferisasi do you have a moment to look over this before we publish as well?

**Review by jenniferisasi — APPROVED** [review]
👍  It looks good to me! Congrats

**Review by svmelton — APPROVED** [review]

**Comment by svmelton** [comment]
Looks great!

**Comment by hawc2** [comment]
Thanks all! I'm publishing the lesson now

**Comment by hawc2** [comment]
@svmelton @ZoeLeBlanc I'm perplexed by what happened here. This pull request was ready to go. I made the mistake of thinking I'd pressed Merge, but then Closed the ticket before finishing merging. 

When I reopened the ticket, the build process ran again, and now the htmlproofer is throwing an error over a missing link. I don't really get how this link suddenly became an issue. What do we do in this situation?

**Comment by jenniferisasi** [comment]
Since I get all the emails... I can help. @hawc2 this is totally normal :) It simply caught a broken link on your second time, because it checks every link we have on the site every time we build a PR. So if that link broke right at the moment you closed the ticket, coincidentally, the build picked it. 

The link belongs to another lesson: introduction-to-populating-a-website-with-api-data.html 

What we usually do in this case: the editor itself tries to find an alternative and replace the link in this same PR. Sometimes we even delete the link completely if impossible to find a solution. 

Hope that helps. Bottomline: totally normal to happen.

**Comment by hawc2** [comment]
Great, thanks so much @jenniferisasi for explaining!

**Comment by hawc2** [comment]
@jenniferisasi in this case, the link throwing the html-error seems to work fine for me: https://www.europeana.eu/api/v2/thumbnail-by-url.json?uri=http%3A%2F%2Fuurl.kbr.be%2F1017835%2Fthumbs%2Fs&size=LARGE&type=IMAGE

If the website renders correctly, what am I supposed to do? This wouldn't be an easy image to replace as its integral to a substantial portion of the lesson it seems

**Comment by jenniferisasi** [comment]
> @jenniferisasi in this case, the link throwing the html-error seems to work fine for me: https://www.europeana.eu/api/v2/thumbnail-by-url.json?uri=http%3A%2F%2Fuurl.kbr.be%2F1017835%2Fthumbs%2Fs&size=LARGE&type=IMAGE
> 
Mmm let's re-run the build; this has happened before too... the build will pick a link that is "broken" for a split second and re-running it "fixes" it all...

Update: Bingo! Just a silly one-second-issue that our build picked when you re-opened the PR 😅

**Comment by hawc2** [comment]
ok strange, well looks like we're good to go!

**Pull request merged**